Pal/regression/Hex.c

```diff
@@ -0,0 +1,16 @@
+/* -*- mode:c; c-file-style:"k&r"; c-basic-offset: 4; tab-width:4; indent-tabs-mode:nil; mode:auto-fill; fill-column:78; -*- */
+/* vim: set ts=4 sw=4 et tw=78 fo=cqt wm=0: */
+
+#include "pal.h"
+#include "pal_debug.h"
+#include "api.h"
+#include "hex.h"
+
+int main() {
+  char buf[17];
+  uint32_t x = 0xefbeadde;
+  pal_printf("Hex test 1 is %s\n", hex2str(&x, buf, 17));
```

Comment:
This is UB in C, this test will fail when porting to any non-little-endian architecture or when compiler optimize the code too aggressively. Fix:
```C
char x[] = {0xde, 0xad, 0xbe, 0xef};
```
Same for the code below.

Also: I'd be better to use `sizeof(buf)` instead of the hardcoded 17.